ext/opcache/jit/zend_jit_arm64.dasc

```diff
@@ -967,36 +968,37 @@ static int logical_immediate_p (uint64_t value, uint32_t reg_size)
 ||	}
 |.endmacro
 
-// long_ins should be addition or subtraction.
-|.macro LONG_ADD_SUB_WITH_IMM, long_ins, op1_addr, lval, tmp_reg1, tmp_reg2
-||	ZEND_ASSERT(lval >=0 && lval <= ADD_SUB_IMM);
-||	if (Z_MODE(op1_addr) == IS_MEM_ZVAL) {
-||		if (((uint32_t)(Z_OFFSET(op1_addr))) > LDR_STR_PIMM64) {
-|			LOAD_32BIT_VAL tmp_reg2, Z_OFFSET(op1_addr)
-|			ldr tmp_reg1, [Rx(Z_REG(op1_addr)), tmp_reg2]
-|			long_ins tmp_reg1, tmp_reg1, #lval
-|			str tmp_reg1, [Rx(Z_REG(op1_addr)), tmp_reg2]
+/* Conduct add/sub between value from memory 'addr' and an immediate value 'val', and
+ * the computation result is stored back into 'addr'.
+ * Note: it should be guaranteed that 'val' can be encoded into add/sub instruction. */
+|.macro LONG_ADD_SUB_WITH_IMM, add_sub_ins, addr, val, tmp_reg1, tmp_reg2
+||	ZEND_ASSERT(((int32_t)(val)) > 0 && ((int32_t)(val)) <= ADD_SUB_IMM);
```

Comment:
Why is ``val`` casted to ``int32_t``?

Reply:
As denoted at line 974, `val` can be encoded as the imm field for add/sub instruction, that means, `val` is in the range of `0 ~ 0xfff`. I think casted to `int32_t` is one way to assert this condition.

Reply:
I think, the new assertion is wrong. ``((int32_t)(1LL<<32)) <= ADD_SUB_IMM`` evaluates to ``true``.
It's better to keep the old (valid) assertion or remove it at all.

Reply:
Yes. You're right. I will keep the old assertion. Thanks.

Reply:
Besides, I will revert my update on macro ADD_IP_WITH_CONST, the same issue with this case.